src/instance_info.rs

```diff
@@ -0,0 +1,171 @@
+use crate::{
+	config::{Config, CONFIG},
+	server::RequestExt,
+	utils::Preferences,
+};
+use askama::Template;
+use build_html::{Container, Html, HtmlContainer, Table};
+use hyper::{http::Error, Body, Request, Response};
+use once_cell::sync::Lazy;
+use serde::{Deserialize, Serialize};
+use time::OffsetDateTime;
+
+// This is the local static that is intialized at runtime (technically at
+// the first request *to the instance-info endpoint) and contains the data
+// retrieved from the instance-info endpoint.
+pub(crate) static INSTANCE_INFO: Lazy<InstanceInfo> = Lazy::new(InstanceInfo::new);
+
+/// Handles instance info endpoint
+pub async fn instance_info(req: Request<Body>) -> Result<Response<Body>, String> {
+	// This will retrieve the extension given, or create a new string - which will
+	// simply become the last option, an HTML page.
+	let extension = req.param("extension").unwrap_or(String::new());
+	let response = match extension.as_str() {
+		"yaml" => info_yaml(),
+		"txt" => info_txt(),
+		"json" => info_json(),
+		"html" | _ => info_html(req),
```

Comment:
```suggestion
		"html" => info_html(req),
```

I think we should have a `_` arm that returns our default 404.